### PR TITLE
Add `--bids-filter-file` parameter

### LINKED ISSUE #591: Add a flag that filters out sessions

### LINKED ISSUE #685: filtering by session
## Summary
it would be very useful to be able to filter by session, for datasets like MyConnectome with many sessions for one participant.

## Next steps
This is enabled in fMRIPrep using the `--bids-filter-file` argument, which has the added benefit of being able to filter on a much larger set of features.  Would it be possible to enable that kind of filtering in XCP-D?

### PR DESCRIPTION
**Pull request by tsalo**
Closes #685 and closes #591.

## Changes proposed in this pull request

- Add `--bids-filter-file` parameter to CLI.
- Change `mni_to_t1w` throughout package to `template_to_t1w`.
- Change `t1w_to_mni` throughout package to `t1w_to_template`.

## Documentation that should be reviewed

https://xcp-d--686.org.readthedocs.build/en/686/usage.html#filtering-inputs-with-bids-filter-files

### CODE REVIEW

**Comment by tsalo** [comment]
Our test data aren't really good for testing the filter file, but I've tested locally on different data locally and it worked well.

**Comment by tsalo** [comment]
@mattcieslak would you be willing to review the documentation? [Here's](https://xcp-d--686.org.readthedocs.build/en/686/usage.html#filtering-inputs-with-bids-filter-files) the section that's been modified. I just don't know if the info I've added is detailed/comprehensive enough for users.

File: xcp_d/utils/doc.py [hunk 1]
```diff
@@ -59,17 +59,17 @@
...
 ] = """
-t1w_to_mni : :obj:`str`
+t1w_to_template : :obj:`str`
     Path to the T1w-to-MNI transform file.
     May be "identity", for testing purposes.
 """
 
 docdict[
-    "mni_to_t1w"
+    "template_to_t1w"
```
Comment (mattcieslak) [inline xcp_d/utils/doc.py:70]: This is so cool. Not sure how I never knew about this functionality

File: docs/usage.rst [hunk 2]
```diff
@@ -44,6 +44,54 @@ Command-Line Arguments
...
+
+``xcp_d`` allows users to choose which preprocessed files will be post-processed with the ``--bids-filter-file`` parameter.
+This argument must point to a JSON file, containing filters that will be fed into PyBIDS.
+
+The keys in this JSON file are unique to ``xcp_d``.
+They are our internal terms for different inputs that will be selected from the preprocessed dataset.
+
+``"bold"`` determines which preprocessed BOLD files will be chosen.
+You can set a number of entities here, including "session", "task", "space", "resolution", and "density".
+We recommend NOT setting the datatype, suffix, or file extension in the filter file.
```
Comment (mattcieslak) [inline docs/usage.rst:60]: Maybe we can add a warning block that says we don't recommend adding additional filters to any of the below fields?

File: xcp_d/utils/doc.py (hunk 1 above)
Comment (tsalo) [reply xcp_d/utils/doc.py:70]: I stole it from nilearn

File: docs/usage.rst (hunk 2 above)
Comment (tsalo) [reply docs/usage.rst:60]: Added in [7ae3994](https://github.com/PennLINC/xcp_d/pull/686/commits/7ae3994fe83698861feb242635463e1dbd7cdaad)

**Review by mattcieslak — APPROVED** [review]

**Comment by tsalo** [comment]
Thanks @mattcieslak!

**Pull request merged**